mxcubeweb/core/adapter/beamline_action_adapter.py

```diff
@@ -1,21 +1,39 @@
+from typing import ClassVar
+
 import logging
 from enum import Enum
 
-from mxcubeweb.core.adapter.adapter_base import ActuatorAdapterBase
+from mxcubecore.HardwareObjects.BeamlineActions import BeamlineActions
+from mxcubecore.BaseHardwareObjects import HardwareObject
+
+from mxcubeweb.core.models.configmodels import AdapterResourceHandlerConfigModel
+
+from mxcubeweb.core.adapter.adapter_base import AdapterBase
 from mxcubeweb.core.models.adaptermodels import (
     HOActuatorValueChangeModel,
     NStateModel,
-    StrValueModel,
 )
 
+resource_handler_config = AdapterResourceHandlerConfigModel(
+    commands=["stop"], attributes=["data"]
+)
+
+
+class BeamlineActionAdapter(AdapterBase):
+    SUPPORTED_TYPES: ClassVar[list[object]] = [BeamlineActions]
 
-class BeamlineActionAdapter(ActuatorAdapterBase):
-    def __init__(self, ho, *args, **kwargs):
+    def __init__(
+        self,
+        ho: HardwareObject,
+        role: str,
+        app,
```

Comment:
I can't import MXCUBEApp due to circular import, should we use Type["MXCUBEApp"] in those cases, or do you think we should deal with such cases ?

Reply:
What do you mean? You mean for type hint annotation?

```
if typing.TYPE_CHECK:
    import ...
```

otherwise maybe something like this:

```suggestion
        app: "mxcubeweb....MXCUBEAPP",
```

Reply:
Yes, thats what I meant. 

I noticed that people do `if typing.TYPE_CHECK` and I guess whatever does runtime type checking has functionality to handle that in two passes ?. I have to admit that I find it really awkward , and that Python again bites its own tail. That however a discussion for another time.

How would `"mxcubeweb....MXCUBEAPP", work ?

Reply:
I have not needed to do this in quite a while, but a few times I had to put quotes around the type hint (so that it is a string, something like: `app: "mxcubeweb.app.MXCUBEApplication"`), to avoid things like circular imports. If I remember correctly, I am a bit fuzzy on the details because it was a while ago.

`typing.TYPE_CHECKING` is always `False` except when the code is being interpreted by  type checking tool (like `mypy`) is running. I am not sure it would help for circular imports though.